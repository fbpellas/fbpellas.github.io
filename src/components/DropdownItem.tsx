import * as React from 'react';
import Dropdown from 'react-bootstrap/Dropdown';

export interface DropdownItemProps {
  anchor: string;
  setPageAndClear: (hash: string) => void;
}

const DropdownItem: React.FC<DropdownItemProps> = (props) => {
  const { anchor, setPageAndClear } = props;
  const href = `#${anchor}`;

  return (
    <Dropdown.Item
      href={href}
      onClick={() => {
        setPageAndClear(anchor);
      }}
    >
      About the Author
    </Dropdown.Item>
  );
};

export { DropdownItem };
