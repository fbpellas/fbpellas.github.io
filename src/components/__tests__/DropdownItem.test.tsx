import React from 'react';
import { render } from '@testing-library/react';
import { DropdownItem } from '../DropdownItem';

test('DropdownItem', () => {
  const fn = jest.fn();
  const { container } = render(<DropdownItem anchor="some-anchor" setPageAndClear={fn} />);
  expect(container).toMatchSnapshot();
});
