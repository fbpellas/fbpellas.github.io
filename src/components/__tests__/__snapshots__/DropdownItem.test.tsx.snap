// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownItem 1`] = `
<div>
  <a
    class="dropdown-item"
    href="#some-anchor"
  >
    About the Author
  </a>
</div>
`;
